src/main/java/org/truffleruby/core/thread/ThreadBacktraceLocationNodes.java

```diff
@@ -117,28 +115,10 @@ public int lineno(DynamicObject threadBacktraceLocation) {
     @CoreMethod(names = {"to_s", "inspect"})
     public abstract static class ToSNode extends UnaryCoreMethodNode {
 
-        @TruffleBoundary
         @Specialization
         public DynamicObject toS(DynamicObject threadBacktraceLocation) {
-            final Activation activation = Layouts.THREAD_BACKTRACE_LOCATION.getActivation(threadBacktraceLocation);
-
-            if (activation.getCallNode() == null) {
-                return coreStrings().BACKTRACE_OMITTED_LIMIT.createInstance();
-            }
-
-            final Node callNode = activation.getCallNode();
-
-            final RootNode rootNode = callNode.getRootNode();
-
-            final SourceSection sourceSection = callNode.getEncapsulatingSourceSection();
-
-            return createString(RopeOperations.format(getContext(),
-                    sourceSection.getSource().getName(),
-                    ":",
-                    sourceSection.getStartLine(),
-                    ":in `",
-                    rootNode.getName(),
-                    "'"));
+            final String description = Layouts.THREAD_BACKTRACE_LOCATION.getDescription(threadBacktraceLocation);
+            return createString(StringOperations.encodeRope(description, UTF8Encoding.INSTANCE));
```

Comment:
Unless you have a compelling reason to do it this way, please use `StringNodes.MakeStringNode`.

Reply:
None beyond copy/paste, it's now fixed.